src/System.Text.Encoding/tests/EncodingTestHelpers.cs

```diff
@@ -10,7 +10,26 @@ public static class EncodingHelpers
     {
         private static readonly RandomDataGenerator s_randomGenerator = new RandomDataGenerator();
 
-        public static void GetByteCount(Encoding encoding, string chars, int index, int count, int expected)
+        public static void Encode(Encoding encoding, string chars, int index, int count, byte[] expected)
+        {
+            byte[] fullArray = new byte[expected.Length + 4];
+            for (int i = 0; i < fullArray.Length; i++)
```

Comment:
I think we can still check that inside GetByteCount and GetBytes. let these methods allocate extra space in the output buffer, initializer only the extra space and then test the values there after we do the encoding results.